web/ui/module/codemirror-promql/src/complete/index.ts

```diff
@@ -32,7 +32,16 @@ export interface CompleteConfiguration {
 }
 
 function isPrometheusConfig(remoteConfig: PrometheusConfig | PrometheusClient): remoteConfig is PrometheusConfig {
-  return (remoteConfig as PrometheusConfig).url !== undefined;
+  const cfg = remoteConfig as PrometheusConfig;
```

Comment:
How about instead of the type assertion, we do it as described in https://www.typescriptlang.org/docs/handbook/advanced-types.html#type-guards-and-differentiating-types: `if ('url' in cfg || ...)`.

It also shows the type assertion as a variant, but for that it says "This isn’t the sort of code you would want in your codebase however." - but I'm not 100% sure why it's that horrible either :P

Reply:
I can change if you want, but I don't really know if it will change the thing. Is it for the beauty of the code ?
I mean I don't get what's the benefit with the other syntaxe. It looks quite the same for me :D.

Reply:
Meh, I dunno. I guess it's fine even if they prefer the other way, I also don't see a big deal either way :+1:

Reply:
let's keep like that and then if someone is saying my code is a crap, then probably we will know what's better :D

Reply:
looks like in the book2 it's not bad anymore to do it like that: https://www.typescriptlang.org/docs/handbook/2/narrowing.html#the-in-operator-narrowing

Reply:
https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates

Reply:
> looks like in the book2 it's not bad anymore to do it like that:

Hmm, the section you linked shows only the way I suggested, but it's fine either way I think.